incubator/hnc/pkg/forest/forest.go

```diff
@@ -144,7 +144,12 @@ func (f *Forest) GetNamespaceNames() []string {
 
 type namedNamespaces map[string]*Namespace
 
-// TODO Store source objects by GK in the forest - https://github.com/kubernetes-sigs/multi-tenancy/issues/281
+// These lookups remain as GVK instead of GK as the interface we get out of Unstructured
+// naturally lends ourselves to using GVK. This also gives us more flexibility to handling
```

Comment:
Rather than saying it's because of the interface from Unstructured - which is an implementation detail - I think I'd suggest pointing to the root cause: HNC is a client of the apiserver, and as a client, it needs to pick specific versions both on the reading and writing side. So while we technically don't need to store the version because it will always match what's in the HNC type configuration, we may as well because we need the GVK anyway while both reading and writing.

Does that make sense?

Reply:
Absolutely that's a good point. I'll make the change and update